tests/assets/loadVideo.tests.ts

```diff
@@ -0,0 +1,211 @@
+import { Assets } from '../../src/assets/Assets';
+import { Texture } from '../../src/rendering/renderers/shared/texture/Texture';
+import { waitUntil } from '../utils/async';
+import { getApp } from '../utils/getApp';
+import { basePath } from './basePath';
+
+import type { VideoSource } from '../../src/rendering/renderers/shared/texture/sources/VideoSource';
+
+async function waitUntilVideoIsLoaded(texture: Texture): Promise<void>
+{
+    // we need to wait until the video is loaded
+    await waitUntil(() => (texture.source as VideoSource).isValid);
+}
+
+async function waitUntilOGVVideoIsLoaded(texture: Texture): Promise<void>
+{
+    await waitUntilVideoIsLoaded(texture);
+
+    // 'canplay' doesn't get fired in the test env for ogv, so we push the width and height manually
+    texture.source.width = texture.source.resourceWidth;
+    texture.source.height = texture.source.resourceHeight;
+}
+
+describe('loadVideo', () =>
+{
+    beforeAll(async () =>
+    {
+        jest.setTimeout(10000);
+        // we need the app loaded for these tests
+        await getApp();
+    });
+
+    beforeEach(() =>
+    {
+        // reset the loader
+        Assets.reset();
+    });
+
+    it('should load MP4 assets', async () =>
+    {
+        await Assets.init({
+            basePath,
+        });
+
+        const video = await Assets.load<Texture>('video/white.mp4');
+
+        await waitUntilVideoIsLoaded(video);
```

Comment:
Why are we awaiting for the video to load?

this seems like an unintended breaking change from v7?

Reply:
The tests were failing because the dimensions returned immediately after the `Assets.load` call were giving zeros, however if I logged the resource and checked a few moments later in the console the values were `1` as expected. So essentially the videos were loading and getting correct values, just not immediately after the `await`. For OGV files, it seems that the `canplay` events were not firing at all, however the media had been loaded and had dimensions, our sources (in the test) were not being updated. I can investigate some more using the playground.

Reply:
Have fixed the issue